python/plugins/processing/algs/gdal/fillnodata.py

```diff
@@ -68,9 +68,16 @@ def initAlgorithm(self, config=None):
                                                        type=QgsProcessingParameterNumber.Type.Integer,
                                                        minValue=0,
                                                        defaultValue=0))
-        self.addParameter(QgsProcessingParameterBoolean(self.NO_MASK,
-                                                        self.tr('Do not use the default validity mask for the input band'),
-                                                        defaultValue=False))
+
+        nomask_param = QgsProcessingParameterBoolean(self.NO_MASK,
+                                                     self.tr('Do not use the default validity mask for the input band'),
+                                                     defaultValue=False,
+                                                     optional=True)
+        # The -nomask option is no longer supported since GDAL 3.4.0 https://github.com/OSGeo/gdal/pull/4201
```

Comment:
As mentionned in the GDAL PR, the -nomask option has never done anything. You can completely remove any addition of it.

Reply:
Totally removing the `NO_MASK` parameter would make the algorithm fail in vain even if such parameter is set to `False` in a currently working Python script or in model, thus I chose to flag it as hidden at least for QGIS using GDAL >= 3.4.
Since the `-nomask` option does nothing, I could set the `NO_MASK` parameter always hidden regardless of the GDAL version and also avoid that the `-nomask` argument is added to the console command even for GDAL version <3.4, raising an exception only if the `NO_MASK` parameter is set to `True` with GDAL >= 3.4.

What do you thing @nyalldawson @rouault?

Reply:
Sounds good to me. Maybe for backward compatibility even siglently ignore NO_MASK if it is set even with GDAL >= 3.4 ?

Reply:
@agiudiceandrea did you see the above comment?

Reply:
I was just waiting for a confirmation from you, since the Even's request to silently ignore NO_MASK if it is set even with GDAL >= 3.4 conflicts with your previous request to throw an exception in such case.

Reply:
Ah, I misunderstood. I thought @rouault was saying backport a change which makes no mask do nothing too.

Reply:
So, is it OK to
- always set the NO_MASK parameter hidden regardless of the GDAL version?
- always avoid that the -nomask argument is added to the console command regardless of the GDAL version?
- don't throw an exception if the NO_MASK parameter is set to True when GDAL >= 3.4?

Reply:
Yep, sounds good to me! Happy to defer to @rouault 's understanding of the situation here.

Reply:
@rouault, I've applied your suggestions.